pkg/pool-cow/contracts/CowRouter.sol

```diff
@@ -329,64 +330,62 @@ contract CowRouter is SingletonAuthentication, VaultGuard, ICowRouter {
         uint256[] memory donatedAmounts,
         uint256[] memory feeAmounts
     ) private {
-        uint256[] memory senderAmounts = new uint256[](donatedAmounts.length);
+        // `returnToSenderAmounts` will contain the number of "leftover" tokens that should be returned to the sender.
+        uint256[] memory returnToSenderAmounts = new uint256[](tokens.length);
         for (uint256 i = 0; i < tokens.length; i++) {
             // Credit the sender for tokens received upfront.
-            uint256 availableFunds = transferAmountHints[i];
-            // Add donation funds (donation to pool + fees) to the required funds.
-            uint256 requiredFunds = donatedAmounts[i] + feeAmounts[i];
+            uint256 senderCredit = transferAmountHints[i];
+            // Debit the sender for tokens transferred in for the donation and protocol fees, so that they are not
+            // returned to the sender.
+            uint256 senderDebt = donatedAmounts[i] + feeAmounts[i];
```

Comment:
This is better than my original more conservative renaming suggestions - more comprehensive and clearer all around!

Credit and Debt are total amounts, so it's correct. You might also consider `senderCredits` and `senderDebits` (debit is the individual amount that changes the total debt, and is parallel to a credit). So later you'd have `if (senderCredits < sender Debits)`.

Really either works; I don't feel strongly. Fine to leave as is.